pkg/crdexplorer/crdexplorer.go

```diff
@@ -26,7 +26,7 @@ type CustomResourceExplorer interface {
 	GetReplicas(object *unstructured.Unstructured) (replica int32, replicaRequires *workv1alpha2.ReplicaRequirements, err error)
 
 	// Retain returns the objects that based on the "desired" object but with values retained from the "cluster" object.
-	Retain(desired *unstructured.Unstructured, cluster *unstructured.Unstructured) (retained *unstructured.Unstructured, err error)
+	Retain(desired *unstructured.Unstructured, cluster *unstructured.Unstructured) error
```

Comment:
Why remove the return value?

Reply:
the former functions only have one return value: error, and don't have Object return value.
https://github.com/karmada-io/karmada/blob/b3424e7d32ebf1342afdd8af9a87f0c46cd9c65a/pkg/util/objectwatcher/retain.go#L54